src/libertem/api.py

```diff
@@ -68,6 +66,101 @@
 RoiT = Optional[Union[np.ndarray, 'SparseArray', 'spmatrix', Tuple[int, ...], IterableRoiT]]
 
 
+class ResultGenerator:
+    def __init__(
+        self,
+        task_results: Generator[UDFResults, None, None],
+        runner: UDFRunner,
+        result_iter,
+    ):
+        self._task_results = task_results
+        self._runner = runner
+        self._result_iter = result_iter
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return next(self._task_results)
+
+    def close(self):
+        self._task_results.close()
+        self._result_iter.close()
+
+    def update_parameters_experimental(self, parameters: List[Dict[str, Any]]):
+        """
+        Update parameters while the UDFs are running.
+
+        `parameters` should be a list of dicts, with one dict for each
+        UDF you are running.
+
+        The dicts should only contain items for those parameters you want to
+        update.
+        """
+        logger.debug("ResultGenerator.update_parameters: %s", parameters)
```

Comment:
The function name in the log message is different from the actual function name.

Would introspection be an option to get the class and function name?

Reply:
Good catch, fixed. Introspection is not worth the hassle IMHO - IIRC you can globally configure the log format to include these automatically, if you really want to.